### ISSUE TITLE
false negative when unboxing potentially null value in element accesses of arrays

### ISSUE THREAD

**Issue by theosotr** [eisop/checker-framework#915]
## Command

```
javac \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED \
  -J--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED \
-processor org.checkerframework.checker.nullness.NullnessChecker \
-cp checker-framework-3.42.0-eisop4/checker/dist/checker.jar Test.java
```
 ## File
 
 ```java
import org.checkerframework.checker.nullness.qual.*;

class Test {
  public static void main(String[] args) {
     Integer y = null;
     String x = (new String[]{"val"})[y];
  }
}
```

## Actual behavior

The code passes the checks, but I get a NPE at runtime

```
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "java.lang.Integer.intValue()" because "<local1>" is null
        at Test.main(Test.java:7)
```
 
## Expected behavior

```
Test.java:7: error: [unboxing.of.nullable] unboxing a possibly-null reference y
```

**Comment by wmdietl** [comment]
Thanks for the report!
This is probably the same as typetools 6804. It will get fixed when we pull in a typetools release with the fix.
